src/corner_kick/src/store/sagas/ros.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * This file specifies the saga for ROS
+ */
+
+import { channel, delay } from 'redux-saga';
+import { put, spawn, take, takeEvery, takeLatest } from 'redux-saga/effects';
+import ROSLIB from 'roslib';
+
+import {
+    connected,
+    disconnected,
+    error,
+    newMessage,
+    setNodes,
+    setParams,
+    setServices,
+    setTopics,
+    start,
+    subscribeTopic,
+    unsubscribeTopic,
+} from '../actions/ros';
+
+let ros: ROSLIB.Ros | null = null;
+const rosChannel = channel();
+
+/**
+ * Function first called when the application first starts
+ */
+export default function* init() {
+    // Listen to start actions and start ROS
+    yield takeLatest('ros/START', startROS);
```

Comment:
GREAT catch. That would be a bug. Hopefully our next PR, introducing a unit testing framework, will avoid these kinds of bugs. I modified the code to avoid these kinds of bugs in the future.

Reply:
Now getting the type directly from the action. Changing the type on the action will update sagas accordingly